tests/test_tools.py

```diff
@@ -160,3 +165,65 @@ def test_get_package_info(monkeypatch, inputs, expected):
     )
     actual_metadata = get_package_info(inputs[0], metadata=inputs[1])
     assert actual_metadata == expected
+
+
+@pytest.mark.parametrize(
+    "inputs, expected_mu",
+    [
+        # Test whether the function returns the correct mu
+        (  # C1: No density or packing fraction (only for known material), expect to get mu from database
+            {
+                "sample_composition": "H2O",
+                "energy": 10,
+            },
+            0.5330,
+        ),
+        (  # C2: Packing fraction (=0.5) provided only (only for known material)
+            {
+                "sample_composition": "H2O",
+                "energy": 10,
+                "packing_fraction": 0.5,
+            },
+            0.2665,
+        ),
+        (  # C3: Density provided only, expect to compute mu based on it
+            # 1. Known material
+            {
+                "sample_composition": "H2O",
+                "energy": 10,
+                "density": 0.987,
+            },
+            0.5330,
+        ),
+        (  # 2. Unknown material
+            {
+                "sample_composition": "ZrO2",
+                "energy": 17,
+                "density": 1.009,
+            },
+            1.252,
+        ),
+        (  # C4: Both density and packing fraction are provided, expect to compute mu based on both
```

Comment:
this doesn't make any sense.  It will result in two numbers, not one number.   We probably want to raise an exception if both are provided.